src/cockpit/channels/filesystem.py

```diff
@@ -71,7 +71,17 @@ def do_open(self, options):
         if watch:
             raise ChannelError('not-supported', message='watching is not implemented, use fswatch1')
 
-        for entry in os.scandir(path):
+        try:
+            scan_dir = os.scandir(path)
+        except OSError as error:
+            if isinstance(error, FileNotFoundError):
+                problem = 'not-found'
+            elif isinstance(error, PermissionError):
+                problem = 'access-denied'
```

Comment:
I guess in these two cases, including the formatted strerror is somewhat redundant.

Also: in all cases, including the filename would probably be helpful.

Reply:
The file name is part of the standard exception message actually:

```
❱❱❱ cpf open fslist1 path=dir1/ watch=False : sleep 1 | cpy |cat

{
  "channel": "ch1",
  "command": "close",
  "message": "[Errno 2] No such file or directory: 'dir1/'",
  "problem": "not-found"
}
```
I think that's good enough?